python/ray/tune/registry.py

```diff
@@ -179,8 +198,22 @@ def _make_key(prefix: str, category: str, key: str):
 
 class _Registry:
     def __init__(self, prefix: Optional[str] = None):
+        """If no prefix is given, use runtime context job ID."""
         self._to_flush = {}
-        self._prefix = prefix or uuid.uuid4().hex[:8]
+        self._prefix = prefix
+        self._registered = set()
+        self._atexit_handler_registered = False
+
+    @property
+    def prefix(self):
+        if not self._prefix:
+            self._prefix = ray.get_runtime_context().get_job_id()
+        return self._prefix
+
+    def _register_atexit(self):
+        if not self._atexit_handler_registered:
+            atexit.register(_unregister_all)
+            self._atexit_handler_registered = True
```

Comment:
I guess we're assuming this is only ever used on the driver. Is that true?

Reply:
That's a good point. rllib registers envs/trainers on import. So if a remote process triggers an import, this may be triggered. I've added a check to only register this handler on the driver, which should be safe.

Reply:
Guarding this on the driver only sounds good to me.